core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -30,115 +59,211 @@
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, int... values) {
+        this(fieldName, values != null ? Ints.asList(values) : null);
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, long... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, long... values) {
+        this(fieldName, Longs.asList(values));
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, float... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, float... values) {
+        this(fieldName, Floats.asList(values));
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, double... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, double... values) {
+        this(fieldName, Doubles.asList(values));
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, Object... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, Object... values) {
+        this(fieldName, values != null ? Arrays.asList(values) : null);
     }
 
     /**
-     * A filter for a field based on several terms matching on any of them.
+     * Constructor used for terms query lookup.
      *
-     * @param name   The field name
-     * @param values The terms
+     * @param fieldName The field name
      */
-    public TermsQueryBuilder(String name, Iterable values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName) {
+        this.fieldName = fieldName;
     }
 
     /**
-     * Sets the execution mode for the terms filter. Cane be either "plain", "bool"
-     * "and". Defaults to "plain".
-     * @deprecated elasticsearch now makes better decisions on its own
+     * A filter for a field based on several terms matching on any of them.
+     *
+     * @param fieldName The field name
+     * @param values The terms
      */
-    @Deprecated
-    public TermsQueryBuilder execution(String execution) {
```

Comment:
sorry I think I am mistaken here, looking deeper, I think we might need to remove execution from the builder in master instead given that we do nothing with it. Will do that.